### ISSUE TITLE
"Page hero - tall" component should be made to accomodate longer words

### ISSUE THREAD

**Issue by theadamparker** [undp/design-system#1375]
https://design.undp.org/?path=/story/components-ui-components-hero-page-pageherooption--page-hero-tall-image

In its current for longer words are wrapping. This is not a desirable behavior, and it should be redesigned.

![Screenshot 2024-09-13 at 12 43 28 PM](https://github.com/user-attachments/assets/d3abd287-0340-488e-b002-77812df6ea6e)

**Comment by Jura** [comment]
Agreed solution would be the following: header elements will be observed on load and window resize and their font-size will be recalculated based on the following conditions:
1. Minimum acceptable font-size is 16px for the small screens and 24px for medium and wider.
2. Script should identify the longest word in the string and reduce its font-size so it fits into container without breaking
3. If even with the minimum font-size the longest word still doesn't fit - minimum font-size is used and the word is allowed to be broken arbitrarily by using: `hyphens: auto; overflow-wrap: break-word;`
